src/explorer/LeetCodeTreeDataProvider.ts

```diff
@@ -13,26 +13,37 @@ import { LeetCodeNode } from "./LeetCodeNode";
 
 export class LeetCodeTreeDataProvider implements vscode.TreeDataProvider<LeetCodeNode> {
 
+    private allProblems: Map<string, IProblem>; // maintains the ownership of all problems.
+
     private treeData: {
-        Difficulty: Map<string, IProblem[]>,
-        Tag: Map<string, IProblem[]>,
-        Company: Map<string, IProblem[]>,
-        Favorite: IProblem[],
+        [Category.All]: IProblem[],
+        [Category.Difficulty]: Map<string, IProblem[]>,
+        [Category.Tag]: Map<string, IProblem[]>,
+        [Category.Company]: Map<string, IProblem[]>,
+        [Category.Favorite]: IProblem[],
     };
 
-    private onDidChangeTreeDataEvent: vscode.EventEmitter<any> = new vscode.EventEmitter<any>();
+    private onDidChangeTreeDataEvent: vscode.EventEmitter<LeetCodeNode> = new vscode.EventEmitter<LeetCodeNode>();
```

Comment:
We can also support refresh the whole explorer, so the parameter type can be written as `LeetCodeNode | null | undefined`

Reply:
The event fire's parameter is `fire(data: T?)` where T=LeetCodeNode, so no need to append null and undefined here